cmd/argo/commands/lint.go

```diff
@@ -3,60 +3,96 @@ package commands
 import (
 	"fmt"
 	"os"
+	"strings"
 
 	"github.com/argoproj/pkg/errors"
 	"github.com/spf13/cobra"
 
 	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/client"
 	"github.com/argoproj/argo-workflows/v3/cmd/argo/lint"
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient"
 )
 
 func NewLintCommand() *cobra.Command {
 	var (
-		strict   bool
-		allKinds bool
-		format   string
+		strict    bool
+		lintKinds []string
+		output    string
 	)
 
+	allKinds := []string{"workflow", "workflow-template", "cron-workflow", "cluster-workflow-template"}
+
 	command := &cobra.Command{
 		Use:   "lint FILE...",
-		Short: "validate files or directories of workflow manifests",
+		Short: "validate files or directories of manifests",
+		Example: `# Lint all manifests in a specified directory:
+
+  argo lint ./manifests
+
+# Lint only manifests of kinds Workflow and CronWorkflow from stdin:
+
+  cat manifests.yaml | argo lint --kinds=wf,cwf -
+`,
 		Run: func(cmd *cobra.Command, args []string) {
 			if len(args) == 0 {
 				cmd.HelpFunc()(cmd, args)
 				os.Exit(1)
 			}
-			ctx, apiClient := client.NewAPIClient()
-			clients := lint.ServiceClients{
-				WorkflowsClient: apiClient.NewWorkflowServiceClient(),
-			}
-			if allKinds {
-				clients.WorkflowTemplatesClient = apiClient.NewWorkflowTemplateServiceClient()
-				clients.CronWorkflowsClient = apiClient.NewCronWorkflowServiceClient()
-				clients.ClusterWorkflowTemplateClient = apiClient.NewClusterWorkflowTemplateServiceClient()
-			}
-			fmtr, err := lint.GetFormatter(format)
-			errors.CheckError(err)
-
-			res, err := lint.Lint(ctx, &lint.LintOptions{
-				ServiceClients:   clients,
-				Files:            args,
-				Strict:           strict,
-				DefaultNamespace: client.Namespace(),
-				Formatter:        fmtr,
-			})
-			errors.CheckError(err)
-
-			fmt.Print(res.Msg())
-			if !res.Success {
-				os.Exit(1)
+
+			if len(lintKinds) == 0 || strings.Contains(strings.Join(lintKinds, ","), "all") {
+				lintKinds = allKinds
 			}
+			RunLint(args, lintKinds, output, strict)
 		},
 	}
 
-	command.Flags().BoolVar(&allKinds, "all-kinds", false, "Lint all kinds, not just workflows")
-	command.Flags().StringVar(&format, "format", "pretty", "Linting results output format. One of: pretty|simple")
+	command.Flags().StringSliceVar(&lintKinds, "kinds", []string{"all"}, fmt.Sprintf("Which kinds will be linted. Can be: %s", strings.Join(allKinds, "|")))
+	command.Flags().StringVarP(&output, "output", "o", "pretty", "Linting results output format. One of: pretty|simple")
 	command.Flags().BoolVar(&strict, "strict", true, "Perform strict workflow validation")
 
 	return command
 }
+
+func getLintClients(client apiclient.Client, kinds []string) (lint.ServiceClients, error) {
+	res := lint.ServiceClients{}
+	for _, kind := range kinds {
+		switch kind {
+		case "workflow", "wf":
+			res.WorkflowsClient = client.NewWorkflowServiceClient()
+		case "workflow-template", "wft":
```

Comment:
these look wrong to me, I don't think they hyphens in them?
also, I think it it `wftmpl`? check the CRD YAML please